### PR TITLE
Make raw data optional in API Requests

### PR DESCRIPTION
**Pull request by WhoSoup**
I was analyzing D/EC blocks for an unrelated issue and noticed that the API will always send both the json data and the raw message, even if you don't need both.  This wastes quite a bit of bandwidth particularly for EC blocks (they're huge due to PegNet). There already is a "get-raw" endpoint for those that only need the raw data but no way to get just the json without the raw.

This changes the following endpoints:
* admin-block
* factoid-block
* entrycredit-block
* dblock-by-height
* ecblock-by-height
* fblock-by-height
* ablock-by-height

("directory-block" itself doesn't even have a raw option, oddly enough)

It adds an optional "noraw" argument to the json call that accepts either "true" or "false". If "noraw" is not present, everything works as before. If "noraw" is true, the API won't return the raw data.

Benchmarked on a LAN copy of factomd, trying to download 3,000 ecblocks  (240650 to 243650):

|  | Time | Size | 
|---|---|---|
|w/ raw|18.6817609s|286.29941082 Mebibyte|
|no raw|13.887978s|160.519980431 Mebibyte|

A ~44% reduction in size and 26% reduction in transfer duration.

### CODE REVIEW

File: wsapi/wsapiStructs.go [hunk 1]
```diff
@@ -273,6 +273,7 @@ type AddressRequest struct {
 
 type HeightRequest struct {
```
Comment (PaulSnow) [inline wsapi/wsapiStructs.go:274]: Consider adding a function:

func (h HeightRequest) IncludeRaw() bool {
    return !h.NoRaw 
}

The problem is that booleans that are the negation of what you need can be very confusing.

File: wsapi/wsapiStructs.go [hunk 2]
```diff
@@ -299,6 +300,7 @@ type HashRequest struct {
 
 type KeyMRRequest struct {
```
Comment (PaulSnow) [inline wsapi/wsapiStructs.go:301]: Same function request

**Comment by WhoSoup** [comment]
Added both suggestions

**Comment by PaulBernier** [comment]
This is looking good, can you rebase the PR against wax rollup @WhoSoup

**Comment by WhoSoup** [comment]
It's rebased. The test that fails is in the p2p package and unrelated to this PR. I'll investigate it in a separate PR.

File: wsapi/wsapiStructs.go [hunk 3]
```diff
@@ -267,6 +267,16 @@ type AddressRequest struct {
...
+}
+
+func (hr *HeightRequest) IncludeRaw() bool {
+	return !hr.NoRaw
+}
+
+type ReplayRequest struct {
+	StartHeight uint32 `json:"startheight"`
+	EndHeight   uint32 `json:"endheight,omitempty"`
 }
```
Comment (PaulBernier) [inline wsapi/wsapiStructs.go:280]: What is this struct doing in this PR? Is that a glitch of the rebase?

Comment (WhoSoup) [reply wsapi/wsapiStructs.go]: Apparently so. My apologies. It's removed.

**Pull request merged**